src/anemoi/training/data/dataset.py

```diff
@@ -290,6 +293,10 @@ def __repr__(self) -> str:
             Timeincrement: {self.timeincrement}
         """
 
+    def __len__(self) -> int:
```

Comment:
We have encountered problems with setting a `__len__` attribute on the `IterableDataset` before. Is this thoroughly tested?

Reply:
Have been using this for a while, did not encounter issues so far. You are welcome to test it out!